### PR TITLE
Use InputStreamSupplier for StreamPayload

### PR DESCRIPTION
**Pull request by jdillon**

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF370-2 :white_check_mark:

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java [hunk 1]
```diff
@@ -70,4 +71,20 @@ public String toString() {
...
+
+  // NOTE: Not using Guava Supplier, as we need to express throws IOException
+
+  /**
+   * Supplies an {@link InputStream}.
+   */
+  public static interface InputStreamSupplier
+  {
+    @NotNull
+    @Nonnull
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java]: Any particular reason we need both of these? Can we standardise on one annotation?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java [hunk 2]
```diff
@@ -30,25 +32,24 @@
 public class StreamPayload
     implements Payload
 {
-  private final InputStream stream;
+  private final InputStreamSupplier stream;
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java]: minor: rename field+parameter to match new type?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java [hunk 3]
```diff
@@ -30,25 +32,24 @@
...
     this.stream = checkNotNull(stream);
     this.size = size;
     this.contentType = contentType;
   }
 
+  /**
+   * Returnes opened stream from configured supplier.
+   */
   @Override
   public synchronized InputStream openInputStream() throws IOException {
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java]: Can we drop `synchronized` now we're not tracking the state anymore?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java [hunk 4]
```diff
@@ -70,4 +71,20 @@ public String toString() {
...
+  //
+  // Supplier
+  //
+
+  // NOTE: Not using Guava Supplier, as we need to express throws IOException
+
+  /**
+   * Supplies an {@link InputStream}.
+   */
+  public static interface InputStreamSupplier
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java]: `static` not needed for interface declarations :)

File: plugins/basic/nexus-repository-raw/src/main/java/org/sonatype/nexus/repository/raw/internal/RawContentPayloadMarshaller.java [hunk 5]
```diff
@@ -57,10 +59,18 @@ public DateTime getLastUpdated() {
     };
   }
 
-  public static Payload toPayload(RawContent content) throws IOException {
+  public static Payload toPayload(final RawContent content) throws IOException {
     checkNotNull(content);
-    return new StreamPayload(content.openInputStream(),
+    return new StreamPayload(
+        new InputStreamSupplier() {
+          @NotNull
```
Comment (mcculls) [inline plugins/basic/nexus-repository-raw/src/main/java/org/sonatype/nexus/repository/raw/internal/RawContentPayloadMarshaller.java]: The change in the nuget format above uses `@Nonnull` whereas this uses `@NotNull`...

**Comment by mcculls** [comment]
+1 ... but we really need to choose a standard annotation for declaring something is not null

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java]: No reason, didn't notice both were here.

**Comment by jdillon** [comment]
@mcculls didn't mean to use the jetbrains annotation, should only need javax.annotation.Nonnull.

We might want to remove the jetbrains annotation library, its really only here for NonNls use which I was keen on at one point, but don't seem to bother with anymore.

**Comment by mrprescott** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**